### DISCUSSION TITLE
Contact attribute value in contact flow is not saving or being passed to Salesforce

### DISCUSSION THREAD

**Discussion by chrisansotti** [amazon-connect/amazon-connect-salesforce-scv#127] [Q&A] [answered]
Goal: I'm trying to pass the option that is selected in the Amazon Connect IVR into a custom field on the Voice Call object in Salesforce. We are using Service Cloud Voice with Partner Telephony by Amazon Connect.

(This is set up exactly as specified at https://developer.salesforce.com/docs/atlas.en-us.voice_developer_guide.meta/voice_developer_guide/voice_example_interactive_voice_response.htm)

I have a custom field in Salesforce on the Voice Call object with an API name of PhoneSupportSelection__c. In the Amazon Connect inbound contact flow I have a "Set Contact Attributes" block that is set up as:

Destination Type: User Defined
Destination Attribute: sfdc-PhoneSupportSelection__c
Text Value: Dashboard

However, the value never passes to Salesforce and when I check the Cloudwatch logs Parameters.Key is listed as PhoneSupportSelection, but there is no value listed.

**Comment by dougjaso** [comment]
Make sure you are setting the attribute before you create the voiceCall. Otherwise, it will only pick this up on CTR data sync.

**Reply by chrisansotti** [reply]
It is before the lambda that creates the voice call. My flow order is:

Main Menu -> Select Tech Support -> Set Queue -> Check Business Hours -> Support sub-menu -> **Set contact attribute based on selection** (input options made via a Lex bot, I've tried button press and voice prompt to no avail) -> Check Staffing -> Check Queue Status -> Play estimated wait time -> **Lambda that creates the voice call**

Does the Lambda need to be moved earlier in the process? I wouldn't think that would matter, but not sure if any of the "Check" steps in between somehow wipe the attribute value.

**Comment by dcarbonetti** [comment]
Your api names don't match

If the sf field is Phone_Support_Selection__c then your contact attribute should be sfdc-Phone_Support_Selection__c

Hope it helps but let us know if there are still issues

**Reply by chrisansotti** [reply]
Sorry, that's a typo on my part. The original API name was Phone_Support_Selection__c but when that wasn't working I removed the underscores in case they weren't valid characters for an attribute key.

**Comment by chrisansotti** [comment, marked as answer]
Turns out I copy/pasted the custom field name from Salesforce and the formatting of the text caused the attribute block to fail. Once I typed it manually it worked fine.
